shbin.py

```diff
@@ -99,21 +115,42 @@ def download(url_or_path, repo, user):
     $ shbin dl https://github.com/Shiphero/pastebin/blob/main/bibo/AWS_API_fullfilment_methods/
     $ shbin dl bibo/AWS_API_fullfilment_methods/
     """
-    path = re.sub(rf"^https://github\.com/{repo.full_name}/(blob|tree)/{repo.default_branch}/", "", url_or_path)
+    full_name_repo = repo.full_name
+    pattern = r"https://github.com/(.*?)/(tree|blob)/"
+    result = re.findall(pattern, url_or_path)
+    if result:
+        full_name_repo = result[0][0]
+
+    is_from_my_repo = "https://" not in url_or_path or full_name_repo == repo.full_name
+    if is_from_my_repo:
+        path = re.sub(rf"^https://github\.com/{repo.full_name}/(blob|tree)/{repo.default_branch}/", "", url_or_path)
+    else:
+        repo = get_repo(full_name_repo)
+        path = get_path(url_or_path)
```

Comment:
if we are trying to parse with a regex, couldn't we generalize it an accept urls or `user/repo/path`   forms for any repo and then check if it's the our not? 

suppose  `shbin zauberzeug/nicegui/examples`